org.eclipse.lyo.oslc4j.codegenerator/src/org/eclipse/lyo/oslc4j/codegenerator/files/generateServiceProviderCatalogSingleton.mtl

```diff
@@ -160,60 +160,39 @@ public class [javaClassNameForSingleton(aServiceProviderCatalog) /]
         }
     }
 
-    [for (aServiceProvider: ServiceProvider | aServiceProviderCatalog.serviceProviders)]
-    public static boolean contains[aServiceProvider.javaName(true)/](final [javaClassName(aServiceProvider) /] serviceProviderInfo) {
-        return serviceProviders.containsKey([javaClassNameForFactory(aServiceProvider)/].constructIdentifier(serviceProviderInfo));
-    }
-
-    public static ServiceProvider get[aServiceProvider.javaName(true)/](HttpServletRequest httpServletRequest, [aServiceProvider.instanceMethodSignature()/])
-    {
-        ServiceProvider serviceProvider;
-
-        synchronized(serviceProviders)
-        {
-            String identifier = [javaClassNameForFactory(aServiceProvider)/].constructIdentifier([aServiceProvider.instanceMethodParameterList()/]);
-            serviceProvider = serviceProviders.get(identifier);
-
-            //One retry refreshing the service providers
-            if (serviceProvider == null)
-            {
-                getServiceProviders(httpServletRequest);
-                serviceProvider = serviceProviders.get(identifier);
-            }
-        }
-
-        if (serviceProvider != null)
-        {
-            return serviceProvider;
-        }
-
-        throw new WebApplicationException(Status.NOT_FOUND);
+    public static boolean containsServiceProvider(final String identifier) {
+        return serviceProviders.containsKey(identifier);
     }
 
-    public static ServiceProvider create[aServiceProvider.javaName(true)/](final [javaClassName(aServiceProvider) /] serviceProviderInfo) 
-            throws OslcCoreApplicationException, URISyntaxException, IllegalArgumentException {
-        String identifier = [javaClassNameForFactory(aServiceProvider)/].constructIdentifier(serviceProviderInfo);
-        if (contains[aServiceProvider.javaName(true)/](serviceProviderInfo)) {
-            throw new IllegalArgumentException(String.format("The SP '%s' was already registered", identifier));
-        }
-        return [javaClassNameForFactory(aServiceProvider)/].createServiceProvider(serviceProviderInfo);
+    public static boolean containsServiceProvider(final ServiceProvider serviceProvider) {
+        return containsServiceProvider(serviceProvider.getIdentifier());
     }
 
-    public static ServiceProvider register[aServiceProvider.javaName(true)/](final HttpServletRequest httpServletRequest,
+    public static ServiceProvider register(final HttpServletRequest httpServletRequest,
```

Comment:
thanks! this looks really simpler! how about deleting this method then and making the one on L179 public?

Reply:
Done. please check I understood you corectly? 
I also generalised the deregister() method. but it is not tested, since it is not being used. But at least it is there for future use.